WLC/web_endpoint.py

```diff
@@ -21,19 +22,28 @@ def api_upload_image():
 
         height, width, _ = img.shape
         pic = Picture(img, 0, 0, width, height, None)
+        saved, key = save_image_to_azure('pictures', pic)
+
+        if not saved:
+            key = None
 
         code, fixed_code, result, error = CodeExecutor().execute_code_img(pic)
 
-        return json.dumps(['unfixed', code, 'fixed', fixed_code, 'result', str(result), 'error', str(error)])
+        return json.dumps(['unfixed', code, 'fixed', fixed_code, 'result', str(result), 'error', str(error), 'key', key])
     else:
         return render_template('upload_test.html')
 
+
 @app.route("/api/resubmit_code", methods=['POST', 'GET'])
 def api_resubmit_code():
     if request.method == 'POST':
         code = request.form['code']
         result, error = CodeExecutor().execute_code(code)
 
+        data = json.dumps(request.json)
+        if 'key' in data and data['key']:
```

Comment:
`if data.get('key'):` is less confusing.
Get returns the default (none)  if the key is not in the dict.

Reply:
Also, from the json.dumps on the previous line, shouldn't data be a string? Hence data['key'] wouldn't work?